### PR TITLE
iconsur 1.6.2

### PR DESCRIPTION
**Pull request by ghost**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/iconsur.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require "language/node"
```
Comment (carlocab) [inline Formula/iconsur.rb:1]:
```suggestion
require "language/node"

```
Let's try to be consistent with other formulae here.

File: Formula/iconsur.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system bin/"iconsur", "--version"
```
Comment (carlocab) [inline Formula/iconsur.rb:17]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @leejongyoung. Just a few comments.

File: Formula/iconsur.rb (hunk 2 above)
Comment (leejongyoung) [reply Formula/iconsur.rb:17]: Need to enter the account password for the system user
Is it possible?

File: Formula/iconsur.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system "mkdir", "-p", "$HOME/Test.app"
+    system bin/"iconsur", "set ", "$HOME/Test.app", "-k", "AppleDeveloper"
+    system bin/"iconsur", "cache"
+    system bin/"iconsur", "unset", "$HOME/Test.app"
```
Comment (carlocab) [inline Formula/iconsur.rb:21]:
```suggestion
    mkdir testpath/"Test.app"
    system bin/"iconsur", "set ", testpath/"Test.app", "-k", "AppleDeveloper"
    system bin/"iconsur", "cache"
    system bin/"iconsur", "unset", testpath/"Test.app"
```
Let's try this.

Comment (leejongyoung) [reply Formula/iconsur.rb:21]: Thanks i'll try it

**Review by chenrui333 — APPROVED** [review]

**Comment by carlocab** [comment]
Thanks for your first contribution to homebrew/core, @leejongyoung. I hope this is the first of many high-quality contributions to come.

**Pull request closed without merging**